### ISSUE TITLE
Action Scheduler queue saturation on blog 7 causes deadlocks (8 fatal) on jobs table + AS claim query

### ISSUE THREAD

**Issue by chubes4** [Extra-Chill/data-machine#2787]
## Summary

Data Machine is over-saturating the Action Scheduler queue on **events.extrachill.com (blog 7)**, producing MySQL InnoDB deadlocks on two different tables. Over a ~12-day window of the live `debug.log`, there were **125 deadlock events**, **8 of which escalated to uncaught `RuntimeException` fatals**.

**This is not an Action Scheduler bug.** The bundled AS is current (**v3.9.3**, with `SKIP LOCKED` support confirmed in `ActionScheduler_DBStore::claim_actions()`), and it is behaving exactly as designed — throwing on a failed claim so the runner retries on the next tick. The deadlocks happen because DM runs too many step-executions concurrently on blog 7, putting both DM's own jobs table and the AS actions table under simultaneous heavy write contention.

## Evidence (live debug.log, extrachill.com network)

Deadlocks grouped by target table:

| Query target | Count | Fatal? |
|---|---|---|
| `UPDATE c8c_7_datamachine_jobs` | 80 | No (swallowed/retried) |
| `actionscheduler_actions` (claim query) | 25 | **8 fatal** |

- **First deadlock seen:** 2026-06-14 15:27:22 UTC
- **Fatal cluster:** 8 fatals tightly clustered in a ~22 min burst, 2026-06-26 00:04:44 → 00:26:44 UTC
- All 80 jobs-table deadlocks originate in `Jobs->start_job()`:

```
UPDATE `c8c_7_datamachine_jobs` SET `status` = 'processing' WHERE `job_id` = NNNN
  made by do_action('wp_ajax_nopriv_as_async_request_queue_runner')
  → ActionScheduler_QueueRunner->do_batch
  → ActionScheduler_Action->execute
  → do_action_ref_array('datamachine_execute_step')
  → DataMachine\Abilities\Engine\ExecuteStepAbility->execute
  → DataMachine\Core\Database\Jobs\Jobs->start_job
```

- The 8 fatals all originate in the AS claim path:

```
Uncaught RuntimeException: Unable to claim actions. Database error: Deadlock found
when trying to get lock; try restarting transaction.
in vendor/woocommerce/action-scheduler/classes/data-stores/ActionScheduler_DBStore.php:1019
  ActionScheduler_DBStore->claim_actions()
  → ActionScheduler_DBStore->stake_claim()
  → ActionScheduler_QueueRunner->do_batch
  → admin-ajax.php (as_async_request_queue_runner)
```

## Root cause

Two coupled problems, both owned by Data Machine:

1. **`Jobs->start_job()` has no deadlock retry.** It issues `UPDATE ... SET status='processing'` with no resilience to MySQL error 1213. Under concurrent runners on the same blog, this deadlocks routinely (80×). These are currently swallowed/retried higher up, but DM should handle its own table's deadlocks explicitly with a "try restarting transaction" retry loop — the standard, MySQL-recommended pattern — rather than relying on incidental recovery.

2. **DM saturates the AS queue.** So many runners are racing to stake a claim simultaneously that even AS's `SKIP LOCKED` claim query deadlocks, and 8 of those bubbled up as fatals. This is a symptom of DM's over-aggressive concurrent fan-out on blog 7, not an AS defect.

## Proposed fix (DM-side, no vendor edits)

1. **Add deadlock-retry to `Jobs->start_job()`** (and any other hot `UPDATE` on `datamachine_jobs`): detect MySQL 1213/1205 in `$wpdb->last_error` and retry the transaction a bounded number of times with small backoff before giving up.
2. **Reduce queue concurrency / stagger dispatch** so fewer runners contend for the same claim window. Options: lower concurrent batches via `action_scheduler_queue_runner_concurrent_batches` and/or batch size, or stagger step scheduling so a single flow doesn't fan out N simultaneous claimable actions.
3. Confirm whether DM should cap concurrent in-flight jobs per blog to avoid this contention pattern recurring on high-volume sites (blog 7 / events is the current hotspot).

## Explicitly NOT in scope

- No edits to `vendor/woocommerce/action-scheduler/` — it is a Composer dependency, is current (3.9.3), and is behaving correctly.
- No AS version bump needed; SKIP LOCKED is already present.

## Operational note

`debug.log` is currently ~118 MB. Worth confirming AS table retention (`wp datamachine retention run`) is keeping `actionscheduler_actions` trimmed, since a bloated actions table worsens claim-query lock contention.
